lib/wordcount.rb

```diff
@@ -0,0 +1,38 @@
+def shred line #split lines into words
+  line.split /[\?\.,_:;@#\n\$\^\*\<\>\%\&!{}\+\n\\"\)\(\-— ]+/
+  #personalize regex for 'word' definition
+end
+
+def add_to_list array_of_words, list #adds only new words to hash list
+  array_of_words.each do |word|
+    key = word.upcase.capitalize #downcase to prevent case mattering
+    if list.has_key? key #if word already exists, inc count
+      list["#{key}"]=list["#{key}"]+1
+    else #else create new hash key and value
+      list["#{key}"] = 1
+    end
+  end
+end
+
+#reads file in and returns array for printing
+def generate_list filename
+  input = File.new filename, "r+" #opens file for reading
+  list = Hash.new #creates hash for wordcount list
```

Comment:
You can instantiate Hashes with a default value that is used when a key is not found. Use `Hash.new(0)` here and you won't need to check if a given key is already in the hash when you need to increment its value.